engines/wintermute/ad/ad_entity.cpp

```diff
@@ -838,6 +840,78 @@ bool AdEntity::scCallMethod(ScScript *script, ScStack *stack, ScStack *thisStack
 		return STATUS_OK;
 	}
 
+#if defined(ENABLE_KINJAL) || defined(ENABLE_HEROCRAFT)
+	//////////////////////////////////////////////////////////////////////////
+	// [WME Kinjal 1.4] SetBeforeEntity / SetAfterEntity
+	// Usage at HeroCraft games: ent.SetBeforeEntity("redDuskaEntity")
+	//////////////////////////////////////////////////////////////////////////
+	else if (strcmp(name, "SetBeforeEntity") == 0 || strcmp(name, "SetAfterEntity") == 0) {
+		stack->correctParams(1);
+		const char *layerName = stack->pop()->getString();
+
+		if (strcmp(getName(), layerName) == 0) {
+			warning("%s(%s): source and target has the same name", name, layerName);
+			stack->pushBool(false);
+			return STATUS_OK;
+		}
+
+		for (uint32 i = 0; i < ((AdGame *)_gameRef)->_scene->_layers.size(); i++) {
+			AdLayer *layer = ((AdGame *)_gameRef)->_scene->_layers[i];
+			for (uint32 j = 0; j < layer->_nodes.size(); j++) {
+				if (layer->_nodes[j]->_type == OBJECT_ENTITY && this == layer->_nodes[j]->_entity) {
+					for (uint32 k = 0; k < layer->_nodes.size(); k++) {
+						if (layer->_nodes[k]->_type == OBJECT_ENTITY && strcmp(layer->_nodes[k]->_entity->getName(), layerName) == 0) {
+							if (j < k && strcmp(name, "SetBeforeEntity") == 0) {
+								k--;
+							} else if (j > k && strcmp(name, "SetAfterEntity") == 0) {
+								k++;
+							}
+
+							int32 delta = j <= k ? 1 : -1;
+							AdSceneNode *tmp = layer->_nodes[j];
+							for (int32 x = j; x != (int32)k; x += delta) {
+								layer->_nodes[x] = layer->_nodes[x + delta];
+							}
+							layer->_nodes[k] = tmp;
+
+							stack->pushBool(true);
+							return STATUS_OK;
+						}
+					}
+				}
+			}
+		}
+
+		warning("%s(%s): not found", name, layerName);
+		stack->pushBool(false);
+		return STATUS_OK;
+	}
+
+	//////////////////////////////////////////////////////////////////////////
+	// [WME Kinjal 1.4] GetLayer / GetIndex
+	//////////////////////////////////////////////////////////////////////////
+	else if (strcmp(name, "GetLayer") == 0 || strcmp(name, "GetIndex") == 0) {
+		stack->correctParams(0);
+
+		for (uint32 i = 0; i < ((AdGame *)_gameRef)->_scene->_layers.size(); i++) {
+			AdLayer *layer = ((AdGame *)_gameRef)->_scene->_layers[i];
+			for (uint32 j = 0; j < layer->_nodes.size(); j++) {
+				if (layer->_nodes[j]->_type == OBJECT_ENTITY && this == layer->_nodes[j]->_entity) {
+					if (strcmp(name, "GetLayer") == 0) {
```

Comment:
Same here, there's no reason to do this on each loop iteration

Reply:
Same here, it's done only once, then `return` is called